code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm

```diff
@@ -433,9 +498,14 @@
 
 	X.face_atom(A) //Face towards the target so we don't look silly
 
-	if(!line_of_sight(X, A)) //Need line of sight.
-		to_chat(X, span_xenowarning("We lost line of sight to the target!"))
-		return fail_activate()
+	if(chargeup != -1)
```

Comment:
As above, just use null and check for the var instead of this specific value... and as before, use child proc calls, you keep adding all this stuff to the base procs for no reason, it makes the code messy and adds unneeded vars.